tests/scalers/neo4j/neo4j_test.go

```diff
@@ -0,0 +1,296 @@
+}
+
+const (
+	deploymentTemplate = `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.DeploymentName}}
+  namespace: {{.TestNamespace}}
+  labels:
+    app: {{.DeploymentName}}
+spec:
+  replicas: 0
+  selector:
+    matchLabels:
+      app: {{.DeploymentName}}
+  template:
+    metadata:
+      labels:
+        app: {{.DeploymentName}}
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:alpine
+`
+
+	secretTemplate = `
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{.SecretName}}
+  namespace: {{.TestNamespace}}
+data:
+  username: {{.Username}}
+  password: {{.Password}}
+`
+	triggerAuthTemplate = `apiVersion: keda.sh/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: {{.TriggerAuthName}}
+  namespace: {{.TestNamespace}}
+spec:
+  secretTargetRef:
+  - parameter: password
+    name: {{.SecretName}}
+    key: password
+  - parameter: username
+    name: {{.SecretName}}
+    key: username
+`
+	scaledObjectTemplate = `
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: {{.ScaledObjectName}}
+  namespace: {{.TestNamespace}}
+spec:
+  scaleTargetRef:
+    name: {{.DeploymentName}}
+  minReplicaCount: {{.MinReplicaCount}}
+  maxReplicaCount: {{.MaxReplicaCount}}
+  triggers:
+  - type: neo4j
+    metadata:
+      host: {{.HostName}}
+      protocol: {{.Protocol}}
+      port: "7687"
+      queryValue: "9"
+      query: 'MATCH (n:Person)<-[r:FOLLOWS]-() WHERE n.popularfor IS NOT NULL RETURN n,COUNT(r) order by COUNT(r) desc LIMIT 1'
+      activationQueryValue: "9"
+    authenticationRef:
+      name: {{.TriggerAuthName}}
+`
+)
+
+func CreateKubernetesResourcesNeo4j(t *testing.T, kc *kubernetes.Clientset, nsName string, data interface{}, templates []Template) {
+	KubectlApplyMultipleWithTemplate(t, data, templates)
+}
+
+func TestNeo4jScaler(t *testing.T) {
+	// Create kubernetes resources
+	kc := GetKubernetesClient(t)
+	// setup neo4j
+	CreateNamespace(t, kc, testNamespace)
+	installNeo4j(t)
+
+	data, templates := getTemplateData()
+	CreateKubernetesResourcesNeo4j(t, kc, testNamespace, data, templates)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, minReplicaCount, 60, 2),
+		"replica count should be %d after 3 minutes", minReplicaCount)
+
+	// test scaling
+	testActivation(t, kc, data)
+	testScaleOut(t, kc, data)
+	testScaleIn(t, kc, data)
+
+	// cleanup
+	DeleteKubernetesResources(t, kc, testNamespace, data, templates)
+}
+
+func installNeo4j(t *testing.T) {
+	_, err := ExecuteCommand(fmt.Sprintf("helm repo add neo4j %s", neo4jHelmRepo))
+	assert.NoErrorf(t, err, "cannot execute command - %s", err)
+	_, err = ExecuteCommand("helm repo update")
+	assert.NoErrorf(t, err, "cannot execute command - %s", err)
+	_, err = ExecuteCommand(fmt.Sprintf("helm install --wait %s neo4j/neo4j --namespace %s --set neo4j.name=%s --set neo4j.password=%s --set volumes.data.mode=defaultStorageClass", scalerName, testNamespace, neo4jUser, neo4jPassword))
```

Comment:
Add the assertion here please

Reply:
`helm` is already waiting here

Reply:
yes, but if helm install fails, we want to fail the tests, that's why the assertion is required